test/Microsoft.AspNetCore.SignalR.Client.Tests/HttpConnectionTests.ConnectionLifecycle.cs

```diff
@@ -136,14 +138,36 @@ Task OnTransportStart()
                             transport: new TestTransport(onTransportStart: OnTransportStart)),
                         async (connection, closed) =>
                     {
-                        var actual = await Assert.ThrowsAsync<Exception>(() => connection.StartAsync());
-                        Assert.Same(expected, actual);
+                        Assert.Equal(0, startCounter);
+                        await connection.StartAsync();
+                        Assert.Equal(passThreshold, startCounter);
+                    });
+                }
+            }
 
-                        // Should succeed this time
-                        shouldFail = false;
+            [Fact]
+            public async Task StartThrowsAfterAllTransportsFail()
```

Comment:
Can you add a test to verify the server behavior as well? Verify that a new connection id is created for each attempt and that all of the events run that you expect to be run.